libraries/Microsoft.Bot.Configuration/BotConfiguration.cs

```diff
@@ -84,11 +84,22 @@ public static async Task<BotConfiguration> LoadFromFolderAsync(string folder, st
         }
 
         /// <summary>
-        /// load the configuration from a .bot file.
+        /// Load the bot configuration by looking in a folder and ing the first .bot file in the folder.
+        /// </summary>
+        /// <param name="folder">Folder to look for bot files. </param>
+        /// <param name="secret">Secret to use to encrypt keys. </param>
+        /// <returns>BotConfiguration.</returns>
+        public static BotConfiguration LoadFromFolder(string folder, string secret = null)
+        {
```

Comment:
Please add a null/empty/whitespace check on folder, and throw an ArgumentNullException.

Reply:
Note: I could be persuaded that the check is in the delegated method, and to avoid code duplication that's the right place for the check. If that's the case please put a code-comment here calling that out.

Reply:
Thinking about it, it seems like more robust to just add checks to all public methods. Somebody may change the behavior of the sync method and forget the checks. Any strong opinions? Otherwise I'll just add full checks to all public methods.

Reply:
Happy to iterate on this though.

Reply:
@carlosscastro I'm a big believer in "All public methods, on all public classes, should have every argument validated". The simplicity of that is always worthwhile. I wish there was a stylecop rule for it.